### ISSUE TITLE
`prefer-includes` doesn't work well with boolean arrays

### ISSUE THREAD

**Issue by jonahsnider** [sindresorhus/eslint-plugin-unicorn#1459]
```ts
const array: boolean[] = [true, false, true];

// Implicit
array.some(x => x);
// Explicit
array.some(x => Boolean(x));
```

Both of the `.some()` calls should be replaced with

```js
array.includes(true);
```

but the rule doesn't detect that, only `array.some(x => x === true)` is detected.
